### PR TITLE
Allow test result list filtering on archived patient

### LINKED ISSUE #3567: [BUG] Unable to search for results of archived patients
### Description
On the results page, if a patient has been archived, the archived patient name doesn't appear in the autocorrect and the archived user results cannot be displayed.

Only related to search since results will show up in all results.

### Steps to reproduce

1. Go to results page
2. Click the "Search by name" input field
2. Enter the name of a user that's been archived
3. Observe that the user's name does not show up in the autocomplete

### Expected behavior
Because archived users still have results in our system, archived user names should appear in the autocorrect (or we can allow users to enter a name and hit enter to search).

### PR DESCRIPTION
**Pull request by nathancrtr**
## Related Issue
- Closes #3567 
- Test results for archived patients are included in the test results list, but such patients do not appear as suggestions in the `Filter by name` input field

## Changes Proposed
- The `showDeleted` parameter on `patients()` GraphQL queries was renamed to `includeArchived`
    - The parameter for the relevant resolver and service methods was also renamed to `includeArchived`
- The behavior of the `showDeleted` parameter on the patient search GraphQL query is **changed in-place**
    - More specifically, this parameter would formerly return _only_ archived users when true
    - This now returns archived users _in addition_ to active ones
    - Normally we would never do this as it's a backwards-compatibility no-no!
        - However, no query ever performed by frontend sets that parameter to `true`
        - Therefore we are able to change the behavior of the existing query rather than creating a side-by-side backwards compatibility shim
    - This is `true` **only** for the test result list filter
    - This is `false` for the "add to queue" search, since we do not want to accept new test results for archived patients

### Can search for test results for archived patients
https://user-images.githubusercontent.com/27730981/188238151-b0241bf3-14ee-4ed6-ad20-01db0f77b943.mov

### Can search for active patients
https://user-images.githubusercontent.com/27730981/188238329-bd723c3f-2f6b-4357-a10c-f26713dd70dc.mov

## Checklist for Author and Reviewer
- [ ] **Please confirm** that I am not changing the behavior of any existing patient search queries by the frontend
    - Specifically, we're looking for calls where `showDeleted: true` where the desired behavior is receiving a list of _only_ archived patients
- [ ] Does this introduce any sneaky side effects?

### Design
- [x] Any UI/UX changes have a designer as a reviewer, and changes have been approved
- [x] Any large-scale changes have been deployed to `test`, `dev`, or `pentest` and smoke-tested by both the engineering and design teams

### Content
- [x] Any content changes have been approved by content team

### Support
- [x] Any changes that might generate new support requests have been flagged to the support team
- [x] Any changes to support infrastructure have been demo'd to support team

### Security
- [x] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [x] Any dependencies introduced have been vetted and discussed

### Documentation
- [x] Any changes to the startup configuration have been documented in the README

### CODE REVIEW

**Comment by DanielSass** [comment]
It sounds like e2e tests are fixed on main, would you mind pulling in main and re-running?

**Review by DanielSass — APPROVED** [review]

File: backend/src/main/java/gov/cdc/usds/simplereport/service/PersonService.java [hunk 1]
```diff
@@ -176,7 +179,7 @@ protected Specification<Person> buildPersonMatchFilter(
    * @param facilityId If null, then it means across accessible facilities in the whole organization
    * @param pageOffset Pagination offset is zero based
    * @param pageSize How many results to return, zero will result in the default page size (large)
-   * @param isArchived Default is false. true will ONLY show deleted users
+   * @param includeArchived Default is false. true will ONLY show deleted users
```
Comment (mehansen) [inline backend/src/main/java/gov/cdc/usds/simplereport/service/PersonService.java:182]: I think this `@param` comment needs to be updated to reflect the new behavior. my understanding is true shows all users and false will continue to only show active users, as before

File: backend/src/main/java/gov/cdc/usds/simplereport/service/PersonService.java [hunk 2]
```diff
@@ -145,7 +145,10 @@ protected Specification<Person> buildPersonSearchFilter(
             : Arrays.stream(namePrefixMatch.split("[ ,]")).collect(Collectors.toList());
 
     // build up filter based on params
-    Specification<Person> filter = inCurrentOrganizationFilter().and(isDeletedFilter(isArchived));
+    Specification<Person> filter = inCurrentOrganizationFilter();
+    if (!isArchived) {
```
Comment (mehansen) [inline backend/src/main/java/gov/cdc/usds/simplereport/service/PersonService.java:149]: non-blocking personal opinion nit: this parameter behaves more like an `includeArchived` in this method as well 😅

File: frontend/src/app/testQueue/addToQueue/AddToQueueSearch-patient-search.test.tsx [hunk 3]
```diff
@@ -55,7 +56,9 @@ const mocks = [
   {
     request: {
       query: QUERY_PATIENT,
+      includeArchived: false,
```
Comment (mehansen) [inline frontend/src/app/testQueue/addToQueue/AddToQueueSearch-patient-search.test.tsx:59]: is this supposed to be in this part of the request? I'm surprised this isn't causing something to fail. are we using this mock? 🧐

**Review by mehansen — COMMENTED** [review]
thanks for changing the parameter name to match the behavior! looks like it was probably a little tedious to go through and change it in the tests

File: backend/src/test/java/gov/cdc/usds/simplereport/api/graphql/ApiUserManagementTest.java [hunk 4]
```diff
@@ -1122,7 +1122,6 @@ void setCurrentUserTenantDataAccess_adminUser_success() {
                 .get("setCurrentUserTenantDataAccess");
     assertEquals("ruby@example.com", user.get("email").asText());
     assertEquals(Role.ADMIN, Role.valueOf(user.get("role").asText()));
-    // assertEquals(ADMIN_PERMISSIONS, extractPermissionsFromUser(user));
```
Comment (mehansen) [inline backend/src/test/java/gov/cdc/usds/simplereport/api/graphql/ApiUserManagementTest.java:1125]: 👍🏻👍🏻

File: backend/src/main/java/gov/cdc/usds/simplereport/service/PersonService.java (hunk 1 above)
Comment (nathancrtr) [reply backend/src/main/java/gov/cdc/usds/simplereport/service/PersonService.java]: Thanks! Done in [990411b](https://github.com/CDCgov/prime-simplereport/pull/4229/commits/990411be2ee1fd3f44795b114f5b67518b14e7f8)

File: backend/src/main/java/gov/cdc/usds/simplereport/service/PersonService.java (hunk 2 above)
Comment (nathancrtr) [reply backend/src/main/java/gov/cdc/usds/simplereport/service/PersonService.java]: Very fair point, done in [990411b](https://github.com/CDCgov/prime-simplereport/pull/4229/commits/990411be2ee1fd3f44795b114f5b67518b14e7f8)

**Comment by zdeveloper** [comment]
@nathancrtr I'll be happy to resolve the merge conflict on this, since I caused it. let me know

**Review by zdeveloper — APPROVED** [review]
LGTM! thank you!

**Review by DanielSass — APPROVED** [review]

**Pull request merged**